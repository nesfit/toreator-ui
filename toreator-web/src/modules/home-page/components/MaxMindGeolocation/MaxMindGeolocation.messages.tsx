import {defineMessages} from "react-intl";

const messages = defineMessages({
  from: {
    id: "MaxMindGeolocation.header",
    defaultMessage: "from",
  },
});

export default messages;
