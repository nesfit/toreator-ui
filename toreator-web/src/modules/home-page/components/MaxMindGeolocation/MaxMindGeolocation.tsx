import {includes} from "ramda";
import * as React from "react";
import {FormattedMessage} from "react-intl";
import {Filters} from "../../constants/toreator";
import {InfoLink, Item} from "../Info/Info.styled";
import messages from "./MaxMindGeolocation.messages";

const hiddenFields = ["geolocation source", "timestamp"];

export const parseArraysToList = (
  array: string[][],
  getIpAddressInfo: (object: {}) => void,
) => {
  if (!array || array.length === 0) {
    return null;
  }
  return (
    <ul>
      {array.map(
        (item: any = []) =>
          item.length === 2 &&
          !includes(item[0], hiddenFields) && (
            <li key={item[0]}>
              {item[0]}:{" "}
              {item[0] === "network" ? (
                <InfoLink
                  onClick={() =>
                    getIpAddressInfo({filter: Filters.NONE, id: item[1]})
                  }
                >
                  {item[1]}
                </InfoLink>
              ) : (
                item[1]
              )}
            </li>
          ),
      )}
    </ul>
  );
};

interface MaxMindGeolocationProps {
  maxMindGeolocation?: any;
  children?: React.ReactNode;
  getIpAddressInfo: (object: {}) => void;
}
const MaxMindGeolocation = ({
  maxMindGeolocation = {},
  children,
  getIpAddressInfo,
}: MaxMindGeolocationProps) => {
  const keys = Object.keys(maxMindGeolocation);
  if (!keys || keys.length === 0) {
    return null;
  }
  return (
    <Item>
      {children}
      <ul>
        {keys.map((key: string) => (
          <li key={key}>
            <FormattedMessage {...messages.from} />{" "}
            <InfoLink
              onClick={() =>
                getIpAddressInfo({
                  filter: Filters.TIME,
                  time: `${key}`,
                })
              }
            >
              {key}
            </InfoLink>
            :
            <ul>
              {parseArraysToList(maxMindGeolocation[key], getIpAddressInfo)}
            </ul>
          </li>
        ))}
      </ul>
    </Item>
  );
};

export default MaxMindGeolocation;
