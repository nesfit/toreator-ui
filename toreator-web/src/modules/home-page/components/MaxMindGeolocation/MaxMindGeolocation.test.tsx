import {shallow} from "enzyme";
import * as React from "react";
import MaxMindGeolocation, {parseArraysToList} from "./MaxMindGeolocation";

describe("<MaxMindGeolocation/>", () => {
  test("default", () => {
    const wrapper = shallow(<MaxMindGeolocation getIpAddressInfo={() => ""} />);
    expect(wrapper).toMatchInlineSnapshot(`""`);
  });

  test("parseArraysToList", () => {
    expect(
      parseArraysToList([["p1", "v1"], ["p2", "v2"]], () => ""),
    ).toMatchSnapshot();
  });
});
