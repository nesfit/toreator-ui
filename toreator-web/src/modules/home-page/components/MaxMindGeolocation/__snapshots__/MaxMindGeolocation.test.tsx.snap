// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaxMindGeolocation/> parseArraysToList 1`] = `
<ul>
  <li>
    p1
    :
     
    v1
  </li>
  <li>
    p2
    :
     
    v2
  </li>
</ul>
`;
